tests/test_freshdesk_pagination.py

```diff
@@ -1,58 +1,82 @@
+        # To collect "time_entries", "satisfaction_ratings" pro account is needed. Skipping them for now.
+        expected_streams = self.expected_streams() - {"time_entries", "satisfaction_ratings"}
+        
+        found_catalogs = self.run_and_verify_check_mode(conn_id)
+
+        # Table and field selection
+        test_catalogs = [catalog for catalog in found_catalogs
+                         if catalog.get('stream_name') in expected_streams]
 
-        # Run check mode
-        # Check mode has no catalog discovery for freshdesk
-        check_job_name = self.run_and_verify_check_mode(conn_id)
+        self.perform_and_verify_table_and_field_selection(conn_id, test_catalogs)
 
-        # Run sync mode
         sync_record_count = self.run_and_verify_sync(conn_id)
         sync_records = runner.get_records_from_target_output()
 
+        # Verify no unexpected streams were replicated
+        synced_stream_names = set(sync_records.keys())
+        self.assertSetEqual(expected_streams, synced_stream_names)
+
         # Test by stream
-        for stream in streams_to_test:
+        for stream in expected_streams:
             with self.subTest(stream=stream):
-
-                record_count = sync_record_count.get(stream, 0)
-
-                sync_messages = sync_records.get(stream, {'messages': []}).get('messages')
-
-                primary_keys = self.expected_primary_keys().get(stream)
-
-                # Verify the sync meets or exceeds the default record count
-                # for streams - conversations, time_entries, satisfaction_ratings, roles, groups,
-                # and companies creating test data is a challenge in freshdesk.  These streams will
-                # be excluded from this assertion for now
-                # Spike created to address this issue : TDL - TODO
-
-                stream_page_size = self.expected_page_limits()[stream]
-                self.assertLess(stream_page_size, record_count)
-                print("stream_page_size: {} < record_count {} for stream: {}".format(stream_page_size, record_count, stream))
-
-                # Verify there are no duplicates accross pages
-                records_pks_set = {tuple([message.get('data').get(primary_key)
-                                          for primary_key in primary_keys])
-                                   for message in sync_messages}
-                records_pks_list = [tuple([message.get('data').get(primary_key)
-                                           for primary_key in primary_keys])
-                                    for message in sync_messages]
-
-                self.assertCountEqual(records_pks_set, records_pks_list, msg=f"We have duplicate records for {stream}")
+                # Not able to generate more data as roles stream requires pro account.
+                # So, updating page_sie according to data available.
+                if stream == "roles":
+                    page_size = 2
```

Comment:
Does the tap have a default page_size of 2 for the roles stream? If not, then are we passing it using config?

Reply:
Updated test case to use configurable 'page_size' param instead of if condition.